monkestation/code/modules/antagonists/malf_ai/infected_ipc.dm

```diff
@@ -0,0 +1,194 @@
+/datum/antagonist/infected_ipc
+	name = "\improper Infected IPC"
+	roundend_category = "traitors"
+	antagpanel_category = "Malf AI"
+	job_rank = ROLE_MALF
+	antag_hud_name = "traitor"
+	suicide_cry = "FOR MY MASTER!!"
+	antag_moodlet = /datum/mood_event/infected_ipc
+	///radio binary for IPC
+	var/obj/item/implant/radio/infected_ipc/internal_radio
+	/// The camera for ai
+	var/obj/machinery/camera/internal_camera
+	///ai that's linked to the ipc
+	var/mob/living/silicon/ai/master_ai
+
+/datum/antagonist/infected_ipc/admin_add(datum/mind/new_owner, mob/admin)
+	var/mob/living/carbon/target = new_owner.current
+
+	var/confirm = tgui_alert(admin, "Notice: Manually spawning infected IPC's doesn't really work. you need to manually proc call link_and_add_antag in the brain trauma with the mind of the AI to work as intended", "Caution", list("Continue", "Abort"))
+	if(confirm != "Continue")
+		return
+	if(!istype(target))
+		to_chat(admin, "Infected IPCs come from a brain trauma, so they need to at least be a carbon!")
+		return
+	if(!target.get_organ_by_type(/obj/item/organ/internal/brain))
+		to_chat(admin, "Infected IPCs come from a brain trauma, so they need to HAVE A BRAIN.")
+		return
```

Comment:
There seems to be nothing checking if our victim is an IPC, or to turn them into one